File: Programmers/게임맵최단거리.py
def solution(maps):
    answer = 0
    
    q = []
    q.append([0, 0])
    
    n = len(maps)-1
    m = len(maps[n])-1
    
    while q:
        pos = q.pop(0)
        if pos[0] > 0 and maps[pos[0]-1][pos[1]] == 1:
            maps[pos[0]-1][pos[1]] = maps[pos[0]][pos[1]]+1
            q.append([pos[0]-1, pos[1]])
        if pos[0] < n and maps[pos[0]+1][pos[1]] == 1:
            maps[pos[0]+1][pos[1]] = maps[pos[0]][pos[1]]+1
            q.append([pos[0]+1, pos[1]])
        if pos[1] > 0 and maps[pos[0]][pos[1]-1] == 1:
            maps[pos[0]][pos[1]-1] = maps[pos[0]][pos[1]]+1
            q.append([pos[0], pos[1]-1])
        if pos[1] < m and maps[pos[0]][pos[1]+1] == 1:
            maps[pos[0]][pos[1]+1] = maps[pos[0]][pos[1]]+1
            q.append([pos[0], pos[1]+1])
            
    if maps[n][m] == 1:
        return -1
    return maps[n][m]

print(solution([[1,0,1,1,1],[1,0,1,0,1],[1,0,1,1,1],[1,1,1,0,1],[0,0,0,0,1]]))

File: JavaScript/게임맵최단거리.js
function solution(maps) {
    const answerList = [];
    const stack = [[0, 0, 1]];
    while (stack.length > 0) {
        const [i, j, moves] = stack.shift();
        if (i + 1 === maps.length && j + 1 === maps[i].length) {
            return moves;
        }
        if (i < maps.length - 1 && (maps[i + 1][j] === 1 || maps[i + 1][j] > moves + 1)) {
            stack.push([i + 1, j, moves + 1]);
            maps[i + 1][j] = moves;
        }
        if (j < maps[i].length - 1 && (maps[i][j + 1] === 1 || maps[i][j + 1] > moves + 1)) {
            stack.push([i, j + 1, moves + 1]);
            maps[i][j + 1] = moves;
        }
        if (i > 0 && (maps[i - 1][j] === 1 || maps[i - 1][j] > moves + 1)) {
            stack.push([i - 1, j, moves + 1]);
            maps[i - 1][j] = moves;
        }
        if (j > 0 && (maps[i][j - 1] === 1 || maps[i][j - 1] > moves + 1)) {
            stack.push([i, j - 1, moves + 1]);
            maps[i][j - 1] = moves;
        }
    }
    return -1;
}
